components/remote_settings/src/remotesettings.udl

```diff
@@ -0,0 +1,53 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+[External="rs_client"]
+typedef extern RemoteSettingsConfig;
+
+namespace remotesettings {};
+
+[Custom]
+typedef string RsJsonObject;
+
+dictionary RemoteSettingsResponse {
+    sequence<Record> records;
+    u64 last_modified;
+};
+
+dictionary Record {
+    string id;
+    u64 last_modified;
+    Attachment? attachment;
+    RsJsonObject fields;
+};
+
+dictionary Attachment {
+    string filename;
+    string mimetype;
+    string location;
+    string hash;
+    u64 size;
+};
+
+[Error]
+enum RemoteSettingsError {
+    "ClientError",
+    "JSONError",
+    "FileError",
+    "ParseIntError",
+};
+
+interface RemoteSettings {
```

Comment:
thought: Should other Rust crates depend on this `remote_settings` crate, or use `support/rs_client`?

In my [Suggest Offline prototyping branch](https://github.com/mozilla/application-services/commit/8902fa73227b744a63ade0848adf8c3e710ea708), I added a skeleton for a new Suggest component that—for now—just pulls down the `quicksuggest` collection and collects some stats on it.

At first, I tried depending on this `remote_settings` crate, but couldn't construct a `RemoteSettings` object, because the `RemoteSettingsConfig` struct (I see you've removed that and made it an extern in your latest push 👀) wasn't public. Changing it to `pub use config::RemoteSettingsConfig` fixed that, but then—after a quick fix to resolve attachment URLs; I left a separate comment on `get_attachment` about that—I was curious how to get at the attachment without downloading it to disk.

So I switched to `support/rs_client` instead. That gave me the interface I wanted, to access the attachment's contents in memory, but then I needed to duplicate the `RemoteSettingsResponse`, `Record`, and `Attachment` definitions in my crate, because they're defined in this crate, not in `rs_client`. If you're 👀 at my branch, the dupes are called `SuggestRecordsResponse`, `SuggestRecord`, and `SuggestAttachment` there 😄

I guess I really have three related questions, about how you're envisioning the split between `remote_settings` and `rs_client`: 😅 

* Which crate should my crate—and future crates, if we add more a-s components that pull data from Remote Settings—depend on?
* If `rs_client` is better, how would you feel about lifting the definitions of `RemoteSettingsResponse`, `Record`, and `Attachment` into that crate, and making them externs in the UDL, too?
* If `remote_settings` is better, how would you feel about adding a new method to download an attachment without writing it to a file?

(My gut feel is that adding the types to the lower-level `support/rs_client` feels like a better option, especially since it'll take a dependency on Serde to parse the capabilities response, anyway, and it looks like Nimbus also uses it—but I don't have the full picture that you do about the split between the two crates!)

Reply:
👋 Hi @linabutler, thank you for the feedback! Wanted to respond to some of this now so you didn't think it had been left in complete limbo. I am still slowly working through some iOS build issues with the `extern` UDL type, and unfortunately this has taken a slight backseat to other work, especially while mhammond is on PTO for the next bit. If this slowness is impacting you, please let me know. I can work with my team to try and reprioritize a bit. 

Your intuition about `rs_client` being the crate that other crates ingest is correct, and so I agree that it would be best to move some more of those foundational types into it. In retrospect I probably should've started with that, but I had aimed for the smallest lift when moving the original client out of Nimbus, which expected pure JSON results. 

I do think those updates can be quick, once I get a working example of one extern type. These build issues have been plaguing me for a bit since I don't have a strong understanding of megazording and UniFFI generation steps 😅

Reply:
>These build issues have been plaguing me for a bit since I don't have a strong understanding of megazording and UniFFI generation steps

Hey @MatthewTighe 👋 !! I just wanted to volunteer to help with any build issues you're facing if you're looking to rubber duck with someone from application-services in case there's context we could help provide once this work gets prioritized again

Reply:
🙏 thank you @tarikeshaq, that would be phenomenal! I'm never sure exactly the right place to go for app services questions/help so I appreciate the volunteerism. I'm hoping to get back to this starting next week if you have time then

Reply:
Thanks so much, @MatthewTighe! I've been meaning to reply to your comment for a while (sorry!), and saw you just pushed your latest changes, so now's a perfect time! 😁

Definitely no immediate rush on getting this in. We're very keen to take advantage of your wonderful work for Cross-Platform Suggest, and I know the Search Consolidation project is interested also—but they're still in the planning and discussion phases, and I think it'll be at least a week or two more before we're ready for engineering beyond prototyping.

We'd be totally open to helping you get it over the line if other work takes priority, and I also volunteer to help with build issues, reviews, pairing, and anything you need for this! 😊

> I had aimed for the smallest lift when moving the original client out of Nimbus, which expected pure JSON results.

That makes perfect sense, thanks for the additional context! I'll do a proper review pass now, but I wanted to make sure not to forget to reply to your note.